code/modules/events/wizard/departmentrevolt.dm

```diff
@@ -15,8 +15,14 @@
 /datum/round_event_control/wizard/deprevolt/admin_setup()
 	if(!check_rights(R_FUN))
 		return
-	var/list/options = list("Random", "Uprising of Assistants", "Medical", "Engineering", "Science", "Supply", "Service", "Security")
-	picked_department = input(usr,"Which department should revolt?","Select a department") as null|anything in options
+
+	var/list/options = list()
+	var/list/pickable_departments = subtypesof(/datum/job_department)
+	for(var/datum/job_department/dep as anything in pickable_departments)
+		options[dep.department_name] = dep
+	picked_department = options[(input(usr,"Which department should revolt?","Select a department") as null|anything in options)]
+	if(!picked_department)
+		return //eh just random they dont care
```

Comment:
this should probably note that not entering anything gives you a random department instead of cancelling or whatever

Reply:
sure